pkg/prometheus/promcfg.go

```diff
@@ -2223,3 +2235,92 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	}
 	return cfg
 }
+
+func (cg *ConfigGenerator) generateTracingConfig(tracingConfig *v1.PrometheusTracingConfig) (yaml.MapItem, error) {
+	cfg := yaml.MapSlice{}
+	objMeta := cg.prom.GetObjectMeta()
+
+	if tracingConfig.ClientType != "" {
+		if tracingConfig.ClientType != "grpc" && tracingConfig.ClientType != "http" {
+			return yaml.MapItem{}, errors.New("Invalid ClientType")
+		}
+
+		cfg = append(cfg, yaml.MapItem{
+			Key:   "client_type",
+			Value: tracingConfig.ClientType,
+		})
+	}
+
+	if tracingConfig.Endpoint == "" {
+		return yaml.MapItem{}, errors.New("Endpoint is required and can't be empty")
+	}
+
+	cfg = append(cfg, yaml.MapItem{
+		Key:   "endpoint",
+		Value: tracingConfig.Endpoint,
+	})
+
+	if tracingConfig.SamplingFraction != "" {
+		parsedFraction, err := strconv.ParseFloat(tracingConfig.SamplingFraction, 64)
+		if err != nil {
+			return yaml.MapItem{}, err
+		}
+		if parsedFraction > 1 || parsedFraction < 0 {
+			return yaml.MapItem{}, errors.New("SamplingFraction must be between 0 and 1")
+		}
+
+		cfg = append(cfg, yaml.MapItem{
+			Key:   "sampling_fraction",
+			Value: parsedFraction,
+		})
+	}
+
+	if tracingConfig.Insecure != nil {
+		cfg = append(cfg, yaml.MapItem{
+			Key:   "insecure",
+			Value: tracingConfig.Insecure,
+		})
+	}
+
+	if tracingConfig.Headers != nil {
+		headers := yaml.MapSlice{}
+		for key, value := range tracingConfig.Headers {
+			headers = append(headers, yaml.MapItem{
+				Key:   key,
+				Value: value,
+			})
+		}
+
+		cfg = append(cfg, yaml.MapItem{
+			Key:   "headers",
+			Value: headers,
+		})
+	}
+
+	if tracingConfig.Compression != "" {
+		if tracingConfig.Compression != "gzip" {
+			return yaml.MapItem{}, errors.New("Invalid Compression")
+		}
+
+		cfg = append(cfg, yaml.MapItem{
+			Key:   "compression",
+			Value: tracingConfig.Compression,
+		})
+	}
+
+	if tracingConfig.Timeout != "" {
```

Comment:
I wonder if we can't set default values with kubebuilder annotation as well, then we would need to make all those nil checks 🤔

Reply:
I did it.
Just ensure that we're setting default values same as Prometheus and it helps clean the code a bit.